### ISSUE TITLE
fetch remote when key expired

### ISSUE THREAD

**Issue by ayoyoness** [pixelated/pixelated-user-agent#857]

**Comment by ayoyoness** [comment]
This keeps track of:
- key expiry date extension
- key renewal
- recursive decryption with inactive keys when the active one fails
-  whenever a public key  is queried: it will force fetching from nicknym when the local one is expired.

From leap/bitmask-dev

**Comment by SweetVirginia** [comment]
So, about attached keys, we have an usability issue with that: https://github.com/pixelated/pixelated-user-agent/issues/841 

@tuliocasagrande what are the solutions we have?

**Comment by tuliocasagrande** [comment]
hello @SweetVirginia 

We need somehow to send the new key as an attachment to the user's contacts, so they can update it on their local keyrings. We can improve the heuristics for when to send the key attached, such as detect when the key changes, for example.

@meskio suggested a better solution could be to do autocrypt and attach the key in the headers, but this is not supported by most of the clients yet.

Regardless of what solution we decide, for now, we thought about sending the key attached at every email. This would be fairly easy to implement and it'd guarantee we're replacing old keys.

**Comment by SweetVirginia** [comment]
Thanks @tuliocasagrande 

Just one more question, why it has to be in every email and not the first time the user exchange emails with any contact like we do when the user create their account?

I remember using enigmail and always attaching my keys and that was slowing down a lot my communication, when someone told me there was no need to do that every time my life got easier hehehe.

**Comment by SweetVirginia** [comment]
Oh wait, I re read and I think what I suggested is exactly the first option, which is hard to develop. Right? "We need somehow to send the new key as an attachment to the user's contacts, so they can update it on their local keyrings. We can improve the heuristics for when to send the key attached, such as detect when the key changes, for example."

**Comment by meskio** [comment]
Yes, we have not good place to store if we had sent the lattest version of the key to each recipient address. That is why is easier to attach all the time for now. I'm happy to explore how to store that and iterate it in the future.

Sending the key only once can also have issues, as people might not import it then and months latter when they needed they are not able to find it.

Anyway, all that will be solve if autocrypt gets implemented in most of the clients.
